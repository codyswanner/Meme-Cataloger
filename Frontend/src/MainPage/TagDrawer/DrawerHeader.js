import React from 'react';
import { IconButton, Toolbar, Typography } from '@mui/material';
import EditIcon from '@mui/icons-material/Edit';


export default function DrawerHeader(props) {

    return(
        <Toolbar sx={{ marginTop: {'xs': "5%", 'md': 0}}}>
            <Typography
                sx={{ fontSize: {'xs': 40, 'md': 25} }}>
                Filter by Tag
            </Typography>
            <IconButton
                onClick={props.handleEdit}
                role='switch'
                sx={{
                    marginLeft: 'auto',
                    transform: {'xs': 'scale(1.7)', 'md': 'scale(1.15)'}
                }}>
                <EditIcon/>
            </IconButton>
        </Toolbar>
    );
};
