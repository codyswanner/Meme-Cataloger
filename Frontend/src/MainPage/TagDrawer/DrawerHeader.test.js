import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import DrawerHeader from './DrawerHeader';


describe('DrawerHeader', () => {
    let handleEdit;
    let component;
    beforeEach(() => {
        handleEdit = jest.fn();
        component = render(<DrawerHeader handleEdit={handleEdit}/>);
    });

    test('snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('enables edit mode', () => {
        const editButton = screen.getByRole('button');

        fireEvent(
            editButton,
            new MouseEvent('click', {
                bubbles: true,
                cancelable: true,
            })
        );

        expect(handleEdit).toHaveBeenCalledTimes(1);
    });
});
