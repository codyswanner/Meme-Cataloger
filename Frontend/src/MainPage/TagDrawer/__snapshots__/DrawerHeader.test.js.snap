// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1xnt1ml-MuiToolbar-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-oj3hyj-MuiTypography-root"
        >
          Filter by Tag
        </p>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yq5f9j-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1xnt1ml-MuiToolbar-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-oj3hyj-MuiTypography-root"
      >
        Filter by Tag
      </p>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-yq5f9j-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
